compiler/nativeGen/X86/Instr.hs

```diff
@@ -459,6 +463,8 @@ x86_regUsageOfInstr platform instr
     DELTA   _           -> noUsage
 
     POPCNT _ src dst -> mkRU (use_R src []) [dst]
+    PDEP   _ src dst -> mkRU (use_R src []) [dst]
+    PEXT   _ src dst -> mkRU (use_R src []) [dst]
```

Comment:
I'm pretty sure the above two lines are wrong.  I wrote it that way just to get it to compile.

The `pdep` and `pext` instructions take three operands, a `src` (read) a `mask` (read), and a `dst` (write), yet I've declared that they takes a `src` (read) and a `dst` (write) only.

I'm looking at what options I have:

```
-- 2 operand form; first operand Read; second Written
-- 2 operand form; first operand Read; second Modified
-- 2 operand form; first operand Modified; second Modified
-- 3 operand form; first operand Read; second Modified; third Modified
-- 1 operand form; operand Modified
-- Registers defd when an operand is written.
-- Registers used when an operand is read.
-- Registers used to compute an effective address.
```

None of them really work for me.  The closest match is the 3 operand one:

* 3 operand form; first operand Read; second Modified; third Modified

I believe what I need is:

first operand Read; second Read; third Write

Reply:
In that case perhaps what you want is,
```haskell
PDEP a b c -> mkRU (use_R a $ use_R b) (def_W c)
```
Does this look better?

Reply:
Looks like I was just missing a `[]`

```
PDEP a b c -> mkRU (use_R a $ use_R b []) (def_W c)
```

Thanks for the hint!